test/python/synthesis/test_multi_controlled_synthesis.py

```diff
@@ -0,0 +1,384 @@
+        """
+        qc = QuantumCircuit(num_ctrl_qubits + 1)
+        qc.append(base_gate.control(num_ctrl_qubits), range(num_ctrl_qubits + 1))
+        test_op = Operator(qc).data
+        cop_mat = self.mc_matrix(base_gate, num_ctrl_qubits)
+        self.assertTrue(matrix_equal(cop_mat, test_op))
+
+
+@ddt
+class TestMCSynthesisCounts(QiskitTestCase):
+    """Test gate counts produced by multi-controlled synthesis methods."""
+
+    @data(5, 10, 15)
+    def test_mcx_n_dirty_i15_cx_count(self, num_ctrl_qubits: int):
+        """Test synth_mcx_n_dirty_i15 bound on CX count."""
+        synthesized_circuit = synth_mcx_n_dirty_i15(num_ctrl_qubits)
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the documentation of synth_mcx_n_dirty_i15
+        self.assertLessEqual(cx_count, 8 * num_ctrl_qubits - 6)
+
+    @data(5, 10, 15)
+    def test_mcx_n_clean_m15_cx_count(self, num_ctrl_qubits: int):
+        """Test synth_mcx_n_clean_m15 bound on CX count."""
+        synthesized_circuit = synth_mcx_n_clean_m15(num_ctrl_qubits)
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the documentation of synth_mcx_n_clean_m15
+        self.assertLessEqual(cx_count, 6 * num_ctrl_qubits - 6)
+
+    @data(5, 10, 15)
+    def test_mcx_1_clean_b95_cx_count(self, num_ctrl_qubits: int):
+        """Test synth_mcx_1_clean_b95 bound on CX count."""
+        synthesized_circuit = synth_mcx_1_clean_b95(num_ctrl_qubits)
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the documentation of synth_mcx_1_clean_b95
+        self.assertLessEqual(cx_count, 16 * num_ctrl_qubits - 8)
+
+    def test_c3x_cx_count(self):
+        """Test synth_c3x bound on CX count."""
+        synthesized_circuit = synth_c3x()
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
+        )
+        transpiled_circuit = pm.run(synthesized_circuit)
+        cx_count = transpiled_circuit.count_ops()["cx"]
+        # The bound from the default construction for C3X
+        self.assertLessEqual(cx_count, 14)
+
+    def test_c4x_cx_count(self):
+        """Test synth_c4x bound on CX count."""
+        synthesized_circuit = synth_c4x()
+        pm = generate_preset_pass_manager(
+            optimization_level=0, basis_gates=["u", "cx"], seed_transpiler=12345
```

Comment:
Does the seed matter?

Reply:
I don't think it does, but I think it's best to force everything be fully deterministic just in case.